pkg/cgu/cgu.go

```diff
@@ -26,6 +33,106 @@ type CguBuilder struct {
 	errorMsg string
 }
 
+// NewCguBuilder creates a new instance of CguBuilder.
+func NewCguBuilder(apiClient *clients.Settings, name, nsname string, maxConcurrency int) *CguBuilder {
+	glog.V(100).Infof(
+		"Initializing new CGU structure with the following params: name: %s, nsname: %s, maxConcurrency: %d",
+		name, nsname, maxConcurrency)
+
+	builder := CguBuilder{
+		apiClient: apiClient,
```

Comment:
Please limit the apiClient under builder struct to:
apiClient clientCgu.Interface

after that update NewCguBuilder function with:
apiClient: apiClient.ClientCgu,

Reply:
You'll also need to refactor Create/Delete/Update/Exist function with limited client and remove Get func

Reply:
Updated